Show column descriptions in record page

Shashank-Venkatesh opened the pull request:
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #4766

<!-- Concisely describe what the pull request does. -->

This pull request updates how column descriptions are presented in the **Record (Quickview)** view.

Instead of rendering the description inline, it is now hidden behind an **info icon** and revealed via a **tooltip on hover**.  
The info icon is rendered **only when a description is available**, keeping the UI clean while still allowing users to easily access column context when needed.

---

## Technical details

- Integrated the existing **info icon + tooltip** UI components to display column descriptions
- Removed inline rendering of description text in the Record (Quickview) view
- Tooltip displays the full description content on hover
- Icon is conditionally rendered only when a description exists
- Scoped changes specifically to the Quickview to avoid unintended side effects in other views
- Ensures consistent behavior and styling by reusing shared UI components

---

## Screenshots

**Expected**  
<img width="706" height="341" alt="Expected behavior" src="https://github.com/user-attachments/assets/cb0e9d22-239c-488d-9ef4-f7a57cab7ad3" />

**Changes I made**  
Description hidden behind an info icon and revealed via tooltip on hover.

<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/878219fe-c4aa-4b4a-8359-10d4af46bd1d" />

---

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]: https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

---

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

Commit:
Addition of Description-Feature as per User request

Commit:
Fixing Test cases

Commit:
Fix: Test Cases

Shashank-Venkatesh committed:
Merge branch 'develop' into fix-issue-4766

Anish9901 commented:
Hey @Shashank-Venkatesh, I see that you have a "Developer Certificate of Origin" section in your PR description but the dropdown is empty, can you actually add a certificate and not leave it empty? Also, when you open up a new PR, update the PR description that is automatically populated from our PR template without removing anything.

Shashank-Venkatesh commented:
@Anish9901 for sure from next time I will make sure that they meet  the requirements of PR template. I have done it as you mentioned.

seancolsen reviewed (changes requested):
- Please use the info icon with a circle around it, just like in the table page.
- Make sure the position of the column menu dropdown arrow relative to the column name is not affected by your changes. Before this PR, the dropdown arrow displayed to the right of the column name. Now it's displaying underneath the column name.
- Keep your diff as minimal as possible. For example I see you've added some unnecessary code comments and also removed a large comment block.

In general, I'd like to see more of an eye towards detail. Double-check your work. Make sure it's clean.

Shashank-Venkatesh commented:
Okay @seancolsen I will work on it fix it.

Commit:
Tweaks and Changes made as per Request

Shashank-Venkatesh committed:
Merge branch 'mathesar-foundation:develop' into fix-issue-4766

Shashank-Venkatesh commented:
@seancolsen Thank you for the feedback. I’ve implemented all the points mentioned: the info icon now has a circle around it as in the table page, the column menu dropdown arrow remains correctly positioned to the right of the column name, and I’ve removed all unnecessary comments to keep the codebase clean. I also double-checked everything to ensure the changes are minimal and precise.

With that being said, I’ve also updated the recent changes as an image in the PR description for reference.

Anish9901 commented on mathesar_ui/src/systems/record-view/DirectField.svelte:
```diff
@@ -21,34 +22,23 @@
 
   import RecordStore from './RecordStore';
 
-  /**
-   * This is used to determine whether to display a `NULL` overlay indicator.
-   * For text data types the indicator is important because otherwise the user
-   * has no way to distinguish an empty string from a `NULL` value. But for some
-   * data types (e.g. Date), we can't show the indicator because the input
-   * component already shows placeholder text to guide the user towards the
-   * required formatting.
-   *
-   * TODO: Refactor this logic. Invert control. Each data type should define a
-   * common param to indicate how a NULL value should be displayed over its
-   * input component. Then we should grab onto that param within this component.
-   * The pattern we're currently using is brittle because if we add new data
-   * types we shouldn't need to update this code here.
-   */
```
@Shashank-Venkatesh as Sean mentioned in his earlier review, this shouldn't be removed.

Shashank-Venkatesh replied:
Added the comment

Anish9901 commented on mathesar_ui/src/systems/record-view/DirectField.svelte:
```diff
@@ -119,16 +128,19 @@
               </LinkMenuItem>
             {/if}
           {/if}
+
```
Please remove all unnecessary `\n` such as these, from your changes.

Shashank-Venkatesh replied:
Removed them.

Commit:
Removed unnecessary \n and also added the required comment

Commit:
Test Cases

Commit:
Fixed the Test Cases

seancolsen committed:
Merge branch 'develop' into fix-issue-4766

seancolsen reviewed (changes requested):
This is pretty close to being ready to merge. 

Please fix this regression where long column names don't display well.

- Before this PR:

    <img width="1358" height="728" alt="image" src="https://github.com/user-attachments/assets/f8705ad3-f5eb-4abb-af7d-4211a9a12591" />

    (Long column names are truncated with a tooltip)

- After this PR:

    <img width="1358" height="728" alt="image" src="https://github.com/user-attachments/assets/ef7bdc56-9982-4361-bd9c-29ebafd237a2" />

    (Long column names cause layout overflow)

Commit:
Fix: Fixed Bug and Changes made

Shashank-Venkatesh commented:
@seancolsen reworked on the mentioned bug and also attached the updated work's image in PR description.

seancolsen reviewed (approved):
This is good now. Thanks @Shashank-Venkatesh
